server/src/controllers/Unsubscribe/resolver.ts

```diff
@@ -0,0 +1,73 @@
+import { verify } from 'jsonwebtoken';
+import { Resolver, Arg, Mutation } from 'type-graphql';
+
+import { getConfig } from '../../config';
+import { prisma } from '../../prisma';
+import {
+  UnsubscribeToken,
+  unsubscribeType,
+} from '../../services/UnsubscribeToken';
+
+const unsubscribeFromChapter = async (userId: number, chapterId: number) => {
+  const chapter = await prisma.chapters.findUniqueOrThrow({
+    where: { id: chapterId },
+    include: { events: true },
+  });
+  await prisma.chapter_users.update({
+    data: { subscribed: false },
+    where: { user_id_chapter_id: { chapter_id: chapterId, user_id: userId } },
+  });
+  const onlyUserEventsFromChapter = {
+    AND: [
+      { user_id: userId },
+      { event_id: { in: chapter.events.map(({ id }) => id) } },
+    ],
+  };
+
+  await prisma.event_users.updateMany({
+    data: { subscribed: false },
+    where: onlyUserEventsFromChapter,
+  });
+  await prisma.event_reminders.deleteMany({
+    where: onlyUserEventsFromChapter,
+  });
+};
+
+const unsubscribeFromEvent = async (userId: number, eventId: number) => {
+  const whereCondition = {
+    user_id_event_id: { event_id: eventId, user_id: userId },
+  };
+  const eventUser = await prisma.event_users.findUniqueOrThrow({
+    where: whereCondition,
+    include: { event_reminder: true },
+  });
+  await prisma.event_users.update({
+    data: {
+      subscribed: false,
+      ...(eventUser.event_reminder && { event_reminder: { delete: true } }),
+    },
+    where: whereCondition,
+  });
+};
+
+const typeToUnsubscribe = {
+  [unsubscribeType.Chapter]: unsubscribeFromChapter,
+  [unsubscribeType.Event]: unsubscribeFromEvent,
+};
+
+@Resolver()
+export class UnsubscribeResolver {
+  @Mutation(() => Boolean)
```

Comment:
I vaguely recall there being some issue with Boolean return types... Don't we have to return `{ id }` or something?

Reply:
I'm not remembering anything troublesome with it, there's few other resolvers returning `Boolean`.

Reply:
Okay, cool. Since it's working now, that's fine.